api/src/main/java/org/openmrs/api/impl/EncounterServiceImpl.java

```diff
@@ -217,17 +219,50 @@ public Encounter saveEncounter(Encounter encounter) throws APIException {
 				toAdd.add(newObs);
 			}
 		}
-		
+
+		// remove all obs in toRemove recursively from encounter
+		Set<Obs> seenIt = new LinkedHashSet<>();
```

Comment:
This is just to make sure If there is an obs is added as it's own group member then I should not process that again. Actually I had no such scenarios in my mind(not sure if it is possible as well). I saw a piece of code in `Encounter.java#addObs()`. Then I thought this might be a scenario and can be added as an extra check.